components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/util/EventHookHandlerUtils.java

```diff
@@ -0,0 +1,317 @@
+import org.wso2.identity.event.common.publisher.model.EventPayload;
+import org.wso2.identity.event.common.publisher.model.SecurityEventTokenPayload;
+import org.wso2.identity.webhook.common.event.handler.constant.Constants;
+import org.wso2.identity.webhook.common.event.handler.internal.EventHookHandlerDataHolder;
+import org.wso2.identity.webhook.common.event.handler.model.EventData;
+import org.wso2.identity.webhook.common.event.handler.model.ResourceConfig;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils.CORRELATION_ID_MDC;
+
+/**
+ * This class contains the utility method implementations.
+ */
+public class EventHookHandlerUtils {
+
+    private static final Log log = LogFactory.getLog(EventHookHandlerUtils.class);
+    private volatile ResourceConfig eventSchema = null;
+    private final Object lock = new Object();
+    private static volatile EventHookHandlerUtils instance;
+
+    private EventHookHandlerUtils() {}
+
+    public static EventHookHandlerUtils getInstance() {
+
+        if (instance == null) {
+            synchronized (EventHookHandlerUtils.class) {
+                if (instance == null) {
+                    instance = new EventHookHandlerUtils();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Retrieve event uri.
+     *
+     * @param eventKey  Event key.
+     * @return Event uri string.
+     * @throws IdentityEventServerException If an error occurs.
+     */
+    public String getEventUri(String eventKey) throws IdentityEventServerException {
+
+        try {
+            ResourceConfig eventConfigObject = getEventConfig(eventKey);
+            if (eventConfigObject.getConfigs() != null &&
+                    eventConfigObject.getConfigs().containsKey(Constants.EVENT_CONFIG_SCHEMA_NAME_KEY)) {
+                return (String) eventConfigObject.getConfigs().get(Constants.EVENT_CONFIG_SCHEMA_NAME_KEY);
+            } else {
+                throw new IdentityEventServerException("Event schema not found in the resource event config " +
+                        "for the eventKey: " + eventKey);
+            }
+        } catch (ClassCastException e) {
+            throw new IdentityEventServerException("Error while casting event config at server side", e);
+        }
+    }
+
+    /**
+     * Retrieve the event config.
+     *
+     * @param eventName Event name.
+     * @return Resource config object.
+     * @throws IdentityEventServerException If an error occurs.
+     */
+    private ResourceConfig getEventConfig(String eventName) throws IdentityEventServerException {
```

Comment:
Looks like both eventName and eventKey is used to refer the event. Choose one and use it consistently

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/39adc5d3ada05a48388b42bb7303c9c2498322d2